amethyst_assets/src/store/mod.rs

```diff
@@ -50,24 +50,26 @@ impl<T> AnyStore for T
     }
 }
 
-impl<T> Store for Box<T>
-    where T: AnyStore + ?Sized
+impl<T, S> Store for T
+    where T: ::std::ops::Deref<Target=S>,
+          S: AnyStore + ?Sized
 {
     type Error = BoxedErr;
 
     fn modified(&self, category: &str, id: &str, ext: &str) -> Result<u64, Self::Error> {
-        T::modified(self, category, id, ext)
+        S::modified(self.deref(), category, id, ext)
```

Comment:
I have no preferences here. I change if people want.
`self` would be enough thanks to deref coercion.